instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/jmx/AttributeValueExtractor.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.jmx;
+
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.INFO;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanInfo;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.TabularData;
+
+/**
+ * A class responsible for extracting attribute values from MBeans. Objects of this class are
+ * immutable.
+ */
+public class AttributeValueExtractor implements LabelExtractor {
```

Comment:
There are a variety of classes / methods that are public when they don't need to be. Can you do a pass to minimize the visibility? 

I see there is are `*.conf.*` packages which need to be accessed outside that package. You could remote those packages and merge everything into the top level, or nest them in an `*.internal.*` package which are exempt from compatibility guarantees. (Technically this artifact is still alpha so we can make breaking changes, but still good to follow project conventions now 🙂.)

Reply:
The public constructors and methods are there, because I wanted the YAML parser to be just one of the ways to generate internal representation of the configuration driving the metrics collection. One of the other ways was to use internal API, using the constructors directly. This code has not become a part of the PR, because I wanted to cut on the size of the PR as much as possible (but it can be added in the future).
If we decide that the only way to configure JMX Metric Insight is via YAML parser, some simplification will be possible.